x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/field_config.tsx

```diff
@@ -1251,4 +1281,168 @@ export const FIELD = (readOnly?: boolean): FieldMap => ({
+    }),
+    hidden: (dependencies) => {
+      const [responseBodyIndex] = dependencies || [];
+      return responseBodyIndex === ResponseBodyIndexPolicy.NEVER;
+    },
+    props: (): EuiFieldNumberProps => ({ min: 1, step: 'any', readOnly }),
+    dependencies: [ConfigKey.RESPONSE_BODY_INDEX],
+  },
+  [ConfigKey.IPV4]: {
+    fieldKey: ConfigKey.IPV4, // also controls ipv6
+    component: ComboBox,
+    label: i18n.translate('xpack.synthetics.monitorConfig.ipv4.label', {
+      defaultMessage: 'IP protocols',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.ipv4.helpText', {
+      defaultMessage: 'Specifies which IP procols to use when pinging the remote host.',
+    }),
+    controlled: true,
+    dependencies: [ConfigKey.IPV6],
+    props: ({ field, setValue, dependencies }): EuiComboBoxProps<string> => {
+      const [ipv6] = dependencies;
+      const ipv4 = field?.value;
+      const values: string[] = [];
+      if (ipv4) {
+        values.push('IPv4');
+      }
+      if (ipv6) {
+        values.push('IPv6');
+      }
+      return {
+        options: [
+          {
+            label: 'IPv4',
+          },
+          {
+            label: 'IPv6',
+          },
+        ],
+        selectedOptions: values.map((version) => ({
+          label: version,
+        })),
+        onChange: (updatedValues: Array<EuiComboBoxOptionOption<string>>) => {
+          setValue(
+            ConfigKey.IPV4,
+            updatedValues.some((value) => value.label === 'IPv4')
+          );
+          setValue(
+            ConfigKey.IPV6,
+            updatedValues.some((value) => value.label === 'IPv6')
+          );
+        },
+        isDisabled: readOnly,
+      };
+    },
+  },
+  [ConfigKey.PROXY_HEADERS]: {
+    fieldKey: ConfigKey.PROXY_HEADERS,
+    component: HeaderField,
+    label: i18n.translate('xpack.synthetics.monitorConfig.proxyHeaders.label', {
+      defaultMessage: 'Proxy headers',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.proxyHeaders.helpText', {
+      defaultMessage: 'Additional headers to send to proxies during CONNECT requests.',
+    }),
+    controlled: true,
+    validation: () => ({
+      validate: (headers) => !validateHeaders(headers),
+    }),
+    error: i18n.translate('xpack.synthetics.monitorConfig.proxyHeaders.error', {
+      defaultMessage: 'Header key must be a valid HTTP token.',
+    }),
+    props: (): HeaderFieldProps => ({
+      readOnly,
+    }),
+  },
+  ['check.response.json']: {
+    fieldKey: ConfigKey.RESPONSE_JSON_CHECK,
+    component: KeyValuePairsField,
+    label: i18n.translate('xpack.synthetics.monitorConfig.responseJSON.label', {
+      defaultMessage: 'Check response body contains JSON',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.responseJSON.helpText', {
+      defaultMessage:
+        'A list of expressions executed against the body when parsed as JSON. Body sizes must be less than or equal to 100 MiB',
```

Comment:
```suggestion
        'A list of expressions executed against the body when parsed as JSON. The body size must be less than or equal to 100 MiB.',
```
I rarely see the `MiB` scale, so double checking if that is what we intend? Mebibyte and not Megabyte?

Reply:
I took this directly from the heartbeat docs, so I'm assuming it's correct/intended.